File: src/prod/src/ServiceModel/hosting2/ContainerEventNotification.cpp
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

#include "stdafx.h"

using namespace std;
using namespace Common;
using namespace ServiceModel;
using namespace Hosting2;

ContainerEventNotification::ContainerEventNotification()
    : sinceTime_()
    , untilTime_()
    , eventList_()
{
}

void ContainerEventNotification::WriteTo(TextWriter & w, FormatOptions const &) const
{
    w.Write("ContainerEventNotification { ");

    w.Write("SinceTime={0}, ", sinceTime_);
    w.Write("UntilTime={0}, ", untilTime_);
    w.Write("EventList=[ ");

    for (auto containerEvent : eventList_)
    {
        w.Write("{0} ", containerEvent);
    }

    w.Write("]}");
}

ErrorCode ContainerEventNotification::FromPublicApi(
    FABRIC_CONTAINER_EVENT_NOTIFICATION const & notification)
{
    sinceTime_ = notification.SinceTime;
    untilTime_ = notification.UntilTime;

    auto error = PublicApiHelper::FromPublicApiList<
        ContainerEventDescription,
        FABRIC_CONTAINER_EVENT_DESCRIPTION_LIST>(notification.EventList, eventList_);

    return error;
}

File: src/prod/src/managed/Api/src/System/Fabric/Hosting/Container/ContainerEventNotification.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

namespace System.Fabric
{
    using System;
    using System.Collections.Generic;
    using System.Fabric.Interop;

    internal sealed class ContainerEventNotification
    {
        internal ContainerEventNotification()
        {
        }

        internal UInt64 SinceTime { get; set; }

        internal UInt64 UntilTime { get; set; }

        internal List<ContainerEventDescription> EventList { get; set; }

        internal IntPtr ToNative(PinCollection pin)
        {
            var nativeNotification = new NativeTypes.FABRIC_CONTAINER_EVENT_NOTIFICATION
            {
                SinceTime = this.SinceTime,
                UntilTime = this.UntilTime,
                EventList = ContainerEventDescription.ToNativeList(pin, this.EventList)
            };

            return pin.AddBlittable(nativeNotification);
        }
    }
}